src/actions/pause/mapScopes.js

```diff
@@ -291,6 +284,168 @@ async function findGeneratedBinding(
   };
 }
 
+async function findGeneratedBindingFromPosition(
+  sourceMaps: any,
+  client: any,
+  source: Source,
+  pos: BindingLocation,
+  name: string,
+  type: BindingType,
+  generatedAstBindings: Array<GeneratedBindingLocation>
+) {
+  const gen = await sourceMaps.getGeneratedLocation(pos.start, source);
+  const genEnd = await sourceMaps.getGeneratedLocation(pos.end, source);
+
+  // Since the map takes the closest location, sometimes mapping a
+  // binding's location can point at the start of a binding listed after
+  // it, so we need to make sure it maps to a location that actually has
+  // a size in order to avoid picking up the wrong descriptor.
+  if (gen.line === genEnd.line && gen.column === genEnd.column) {
+    return null;
+  }
+
+  return generatedAstBindings.reduce(async (acc, val) => {
+    const accVal = await acc;
+    if (accVal) {
+      return accVal;
+    }
+
+    const operations = mapBindingToGeneratedRange(
+      val,
+      {
+        start: gen,
+        end: genEnd
+      },
+      type === "import"
+    );
+
+    if (operations) {
+      const hasCall = operations.some(op => op.type === "call");
+      if (hasCall) {
+        // In theory this could fall back to trying to use client.evaluate.
+        return null;
+      }
+
+      let desc = val.desc;
+      if (desc) {
+        for (const op of operations) {
+          if (op.type === "call" || op.type === "inherit") {
+            continue;
+          }
+
+          const objectClient = createObjectClient(desc.value);
+          desc = (await objectClient.getProperty(op.property)).descriptor;
+          if (!desc) {
+            return null;
+          }
+        }
+      }
+
+      return {
+        name: val.name,
+        desc: desc
+      };
+    }
+
+    return null;
+  }, null);
+}
+
+function mapBindingToGeneratedRange(
```

Comment:
can this be commented

hmm... is there a better way to describe returning operations... I wouldnt guess that is what's happening with the method name

Reply:
Good call.